### ISSUE TITLE
UnusedVariable false positive: external array index counter inside loop

### ISSUE THREAD

**Issue by mpesari** [slevomat/coding-standard#606]
Here's an example.

```php
<?php

$a = [];
$i = 0;

foreach ([1, 2, 3] as $x) {
    $a[$i++] = $x;
}
```

```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 7 | ERROR | Unused variable $i.
----------------------------------------------------------------------
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/777ab1871859f5ab5fe3165ad0c2c855b02f5b7c
